File: chat-bot/proxy.js
// Simple HTTP proxy to bypass SSL verification for DashScope API
const https = require('https');
const http = require('http');
const { URL } = require('url');

const DASHSCOPE_BASE = 'https://dashscope.aliyuncs.com';
const PORT = process.argv[2] || 8081;

const server = http.createServer((req, res) => {
    const targetUrl = new URL(req.url, DASHSCOPE_BASE);
    
    console.log(`[PROXY] ${req.method} ${req.url} -> ${targetUrl.href}`);

    // Read request body
    let body = [];
    req.on('data', chunk => body.push(chunk));
    req.on('end', () => {
        const bodyData = Buffer.concat(body);

        // Build headers (exclude hop-by-hop headers)
        const headers = {};
        for (const [key, value] of Object.entries(req.headers)) {
            if (!['host', 'connection', 'transfer-encoding'].includes(key.toLowerCase())) {
                headers[key] = value;
            }
        }

        const options = {
            method: req.method,
            headers,
            rejectUnauthorized: false,
            timeout: 30000
        };

        const proxyReq = https.request(targetUrl, options, (proxyRes) => {
            res.writeHead(proxyRes.statusCode, proxyRes.headers);
            proxyRes.pipe(res);
        });

        proxyReq.on('error', (err) => {
            console.error(`[PROXY] Error: ${err.message}`);
            res.writeHead(502, { 'Content-Type': 'application/json' });
            res.end(JSON.stringify({ error: err.message }));
        });

        proxyReq.on('timeout', () => {
            proxyReq.destroy();
        });

        if (bodyData.length > 0) {
            proxyReq.write(bodyData);
        }
        proxyReq.end();
    });
});

server.listen(PORT, '0.0.0.0', () => {
    console.log(`[PROXY] Server running on port ${PORT} (no SSL verify)`);
});

File: chat-bot/proxy.py
# -*- coding: utf-8 -*-
"""
简单的 HTTP 代理服务器，用于绕过 SSL 证书验证问题
使用 requests 库，禁用环境代理
"""
import http.server
import json
import sys
import urllib3

# 禁用 SSL 警告
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)

try:
    import requests
except ImportError:
    print("Error: requests library not found. Install with: pip install requests")
    sys.exit(1)

DASHSCOPE_BASE_URL = "https://dashscope.aliyuncs.com"

# 创建一个禁用环境代理的 Session
session = requests.Session()
session.trust_env = False  # 禁用环境代理
session.verify = False  # 禁用 SSL 验证

class ProxyHandler(http.server.BaseHTTPRequestHandler):
    def do_GET(self):
        print(f"[PROXY] GET {self.path}")
        self.proxy_request('GET')

    def do_POST(self):
        print(f"[PROXY] POST {self.path}")
        self.proxy_request('POST')

    def proxy_request(self, method):
        url = f"{DASHSCOPE_BASE_URL}{self.path}"
        
        content_length = int(self.headers.get('Content-Length', 0))
        body = self.rfile.read(content_length) if content_length > 0 else None
        
        # 构建请求头
        headers = {}
        for key, value in self.headers.items():
            if key.lower() not in ('host', 'connection', 'transfer-encoding'):
                headers[key] = value
        
        try:
            response = session.request(
                method=method,
                url=url,
                headers=headers,
                data=body,
                timeout=30
            )
            
            self.send_response(response.status_code)
            for key, value in response.headers.items():
                if key.lower() not in ('transfer-encoding', 'connection'):
                    self.send_header(key, value)
            self.end_headers()
            
            self.wfile.write(response.content)
            
        except requests.exceptions.HTTPError as e:
            print(f"[PROXY] HTTP Error: {e}")
            self.send_response(e.response.status_code)
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(e.response.content)
            
        except Exception as e:
            print(f"[PROXY] Error: {e}")
            self.send_response(502)
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({"error": str(e)}).encode())

if __name__ == '__main__':
    port = int(sys.argv[1]) if len(sys.argv) > 1 else 8081
    server = http.server.HTTPServer(('0.0.0.0', port), ProxyHandler)
    print(f"[PROXY] Server running on port {port} (no proxy, no SSL verify)")
    try:
        server.serve_forever()
    except KeyboardInterrupt:
        print("\n[PROXY] Server stopped")
        server.server_close()
